x/dex/keeper/grpc_query_user_deposits.go

```diff
@@ -70,3 +74,14 @@ func (k Keeper) UserDepositsAll(
 		Pagination: pageRes,
 	}, nil
 }
+
+func (k Keeper) addPoolData(ctx sdk.Context, record *types.DepositRecord) *types.DepositRecord {
+	pool, found := k.GetPool(ctx, record.PairId, record.CenterTickIndex, record.Fee)
+	if found {
+		record.Pool = pool
+	}
+	supply := k.bankKeeper.GetSupply(ctx, pool.GetPoolDenom())
```

Comment:
i think it must never happens, where there is no pool found? but still, it looks strange, if no pool found you try to GetPoolDenom from nil pointer. We can change if statement
```
if !found {
return
}
```
otherwise fill the data

Reply:
also probably it makes sense to return an error e.g. "pool data not found" to the calls with `req.IncludePoolData=true` in this case. so the users explicitly asked to get pool data get an error if no data found, and not just a response with no pool data. just a suggestion, not sure how to make it the right way from the user's perspective